pkg/controller/common/manifestprocessing.go

```diff
@@ -40,35 +40,36 @@ func NewManifestProcessor(controllerResources ControllerResources, appInstance,
 	}
 }
 
-func (p *ManifestProcessor) ProcessManifests(manifests []manifest.Manifest, component string) error {
+func (p *ManifestProcessor) ProcessManifests(ctx context.Context, manifests []manifest.Manifest, component string) error {
 	allErrors := []error{}
 
-	origLogger := p.Log
-	defer func() { p.Log = origLogger }()
+	origCtx := ctx
+	origLogger := LogFromContext(ctx)
 	for _, man := range manifests {
-		p.Log = origLogger.WithValues("manifest", man.Name)
+		log := origLogger.WithValues("manifest", man.Name)
+		ctx = NewContextWithLog(origCtx, log)
```

Comment:
`ctx := NewContextWithLog(ctx, LogFromContext(ctx).WithValues("manifest", man.Name))` should do the trick.

And, generally speaking, for all cases where we create a subcontext, should this be followed by, `defer func() { ctx.Done() }()`?  (except in this case, where we're in a loop, which is a bit trickier)

Reply:
No, the local log variable needs to include the "manifest" value, too - there are log statements in this function that need to use the same log as the one we're injecting into the Context. If I use your trick, then I'd have to get the logger from the new context again.

We don't need to call `ctx.Done()`. as it only returns a channel that is used to cancel long-running operations - the Client calls it when performing external calls so that they can be cancelled if someone cancels the context. There is no other function to mark the context "done".